lib/active_model/serializer.rb

```diff
@@ -164,6 +164,14 @@ def json_key
       end
     end
 
+    def id
+      object.id if object
+    end
+
+    def type
+      object.class.to_s.demodulize.underscore.pluralize
```

Comment:
Wouldn't be a good idea to add comments or `GOTCHA`'s to those methods with explanation why the was placed here and what the original intention was behind that decision?